File: leetCode/1528restoreString/solution.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    var result = [];
    for(var i = 0; i < indices.length; i++){
        result[indices[i]] = s.charAt(i);
    }

    return result.join("");
};
console.log(restoreString("codeleet", [4,5,6,7,0,2,1,3]));

File: leetCode/1528restoreString/solution.java
class Solution {
    public String restoreString(String s, int[] indices) {
        //char[] newStr = new char[s.length()];
        char[] result = new char[indices.length];
        for(int i = 0; i < indices.length; i++){
            result[indices[i]] = s.charAt(i);
        }

        return new String(result);
    }
}